src/dcd/server/autocomplete/complete.d

```diff
@@ -322,6 +331,18 @@ AutocompleteResponse parenCompletion(T)(T beforeTokens,
 	return response;
 }
 
+CompletionToken getCompletionToken(T)(T beforeTokens) {
```

Comment:
this function should probably better look at the tokens e.g. to find `[` and `!` earlier than just the last token for things like

```d
foo[bar, |] // should be bracket
foo!(bar, |) // should be bang
```